test/app/service/user.test.js

```diff
@@ -0,0 +1,109 @@
+const { app, mock, assert } = require('egg-mock/bootstrap')
```

Comment:
I didn't find the test case for following/unfollowing, I'm not asking you to add it(it's ok if you don't want).
Just I feel like the implementation with `addToSet` has an issue with duplicated followers, maybe I'm wrong but could you db-check what happens if following the same user twice?

Reply:
Only one record. 

While if you follow, unfollow then follow the same user, there will be two, but one is valid, the other one is invalid. 

This is a cosmic issue per my understanding. On the other hand, it is some kind of follow/unfollow history, may be useful. :)

Reply:
I'd like to challenge you a bit more, let's say, the following/follower is for the latest relationship of users(not the history),
could you try to add some cases in the test code to see if it works?